### ISSUE TITLE
Automated builds from tar archives are not working

### ISSUE THREAD

**Issue by sharkwouter** [pspdev/psp-packages#91]
# Summary
It turns out that the file command is broken, which is used by the makepkg command. This makes makepkg think that tar.gz files are gzip archives, making it use gzip to extract it. The result is that tar files are not extracted fully, causing failures.

# Details
The `makepkg` command uses the following `file` command:
```
file -S -bizL -- "$file"
```
The `file` variable is set to the archive name.

Here is an example in what that results in on alpine:
```
application/x-empty; charset=binary compressed-encoding=application/gzip; charset=binary
```
On Arch and Debian 12 (which use the same version of file) this results in the following with the same file:
```
application/x-tar; charset=binary compressed-encoding=application/gzip; charset=binary
```
This is not fixed on alpine edge.

# Possible solutions

At the moment I see the following options: 

- Fix this in alpine
- Use an older alpine version
- Patch pacman to just try bsdtar if the archive is seen as empty

 I don't think there are many other ways around this one.

**Comment by sharkwouter** [comment]
Here is how the arch package is build: https://gitlab.archlinux.org/archlinux/packaging/packages/file/-/blob/main/PKGBUILD#L33

Here is how the alpine package is build: https://git.alpinelinux.org/aports/tree/main/file/APKBUILD#n25

**Comment by sharkwouter** [comment]
There is a fix for it upstream in file, but it is not available in a new release yet: https://github.com/file/file/commit/1dd21dd360472d7b830825df8e40a06cdc1cbbcf

**Comment by sharkwouter** [comment]
I made an upstream issue: https://gitlab.alpinelinux.org/alpine/aports/-/issues/15047

**Comment by sharkwouter** [comment]
This was resolved upstream now in Alpine 3.18. I'll test it out

**Comment by sharkwouter** [comment]
This has been resolved now thanks to the upstream fix.
